lib/id3c/cli/command/location.py

```diff
@@ -289,3 +292,161 @@ def parse_features(path):
         raise Exception("Unsupported CRS")
 
     return list(collection)
+
+
+@location.command("lookup")
+@click.option("--scale",
+    metavar = "<scale>",
+    default = "tract",
+    show_default = True,
+    help = "Relative size or extent of all locations (e.g. tract, PUMA) to lookup")
+@click.option("--filename",
+    metavar = "<filename.{csv,tsv,xls,xlsx}>",
+    type = click.Path(exists=True))
+@click.option("--lat-column",
+    metavar = "<column>",
+    default = "lat",
+    show_default = True,
+    help = "Column name of latitude")
+@click.option("--lng-column",
+    metavar = "<column>",
+    default = "lng",
+    show_default = True,
+    help = "Column name of longitude")
+@click.option("--drop-input-columns",
+    is_flag = True,
+    help = "Optional flag to drop input columns from the output")
+
+def lookup(scale,
+           lat_column,
+           lng_column,
+           drop_input_columns,
+           filename=None):
+    """
+    Given a <filename.{csv,tsv,xls,xlsx}> that contains latitudes and longitudes,
+    lookup the location within ID3C at the provided <scale>.
+
+    If no <filename.{csv,tsv,xls,xlsx}> is provided, then try to read data from stdin,
+    assuming data formatted as comma-separated values. This is expected when
+    piping output from `id3c geocode` directly into this command.
+
+    Lookup results are output to stdout as comma-separated values, with
+    location identifier as <scale>_identifier. Original input can be dropped
+    with the --drop-input-columns flag.
+    """
+    input_df = load_lookup_input(filename)
+    lat_lngs = extract_lat_lng_from_input(input_df, lat_column, lng_column)
+
+    db = DatabaseSession()
+    locations = []
+
+    for lat_lng in lat_lngs:
+        location = location_lookup(db, lat_lng, scale)
+        locations.append(location.identifier if location else None)
+
+    if drop_input_columns:
+        output_df = pd.DataFrame(locations, columns = [f'{scale}_identifier'])
+    else:
+        output_df = input_df
+        output_df[f'{scale}_identifier'] = locations
```

Comment:
I would prefer to create the `output_df` DataFrames the same way and then to conditionally drop the input columns later. 
Or, conversely, to create the DataFrame without the input columns and re-attach them only if they are requested.